src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -882,6 +900,12 @@ export class BundleDataClient {
                 // At this point, the v3RelayHashes entry already existed meaning that there is a matching deposit,
                 // so this fill is validated.
                 v3RelayHashes[relayDataHash].fill = fill;
+
+                // If the fill is invalid due to relayer repayment addresses, return early.
+                const fillHasValidRepaymentAddress = await verifyFill(fill, destinationClient);
```

Comment:
I might move this check as early as possible into this async () call, because I think its strange that we would save this fill into `v3RelayHashes` if we subsequently invalidate it here. That way also if we overwrite the fill, we can do so as early as possible. For example, we could change to something like `async (_fill) => const fill = fillHasValidRepayment(_fill)` to minimize the diff

Reply:
https://github.com/across-protocol/sdk/pull/842/commits/48aecfd51aa3d43c36757b1942456099238973cc